### PR TITLE
Helio Target

### PR DESCRIPTION
**Pull request by apocolipse**
Helio Spring target

### CODE REVIEW

**Comment by kidBrazil** [comment]
Looks good to me, lets wait for @rs2k for a final word.

File: src/main/drivers/accgyro/accgyro_mpu.c [hunk 1]
```diff
@@ -49,11 +53,14 @@
 #include "drivers/accgyro/accgyro_spi_mpu6000.h"
 #include "drivers/accgyro/accgyro_spi_mpu6500.h"
 #include "drivers/accgyro/accgyro_spi_mpu9250.h"
+#include "drivers/accgyro/accgyro_imuf9001.h"
 #include "drivers/accgyro/accgyro_mpu.h"
 
-
+volatile int dmaSpiGyroDataReady = 0;
```
Comment (rs2k) [inline src/main/drivers/accgyro/accgyro_mpu.c]: this is not used, it's kind of a place holder. Maybe we should remove it for now?

File: src/main/drivers/dma_spi.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+static inline void dmaSpiCsLo(void)
+{
+    gpio_write_pin(DMA_SPI_NSS_PORT, DMA_SPI_NSS_PIN, 0);
+}
+
+static inline void dmaSpiCsHi(void)
+{
+    gpio_write_pin(DMA_SPI_NSS_PORT, DMA_SPI_NSS_PIN, 1);
```
Comment (rs2k) [inline src/main/drivers/dma_spi.c]: these functions probably belong in the gpio driver files, but the bf hal is being used there, so maybe we need a new "fast_gpio" driver

File: src/main/sensors/gyro.c [hunk 3]
```diff
@@ -160,6 +165,35 @@ static void gyroInitSensorFilters(gyroSensor_t *gyroSensor);
 
 PG_REGISTER_WITH_RESET_TEMPLATE(gyroConfig_t, gyroConfig, PG_GYRO_CONFIG, 1);
 
+#ifdef USE_GYRO_IMUF9001
```
Comment (rs2k) [inline src/main/sensors/gyro.c]: dang maros!

File: src/main/interface/cli.c [hunk 4]
```diff
@@ -3767,6 +3780,31 @@ const clicmd_t cmdTable[] = {
 #endif
 };
 
+#ifdef USE_GYRO_IMUF9001
+static void cliImufUpdate(char *cmdline)
+{
+    UNUSED(cmdline);
+    cliPrint("I muff, you muff, we all muff for IMU-F!");
```
Comment (rs2k) [inline src/main/interface/cli.c]: Best. Line. Ever.

File: src/main/target/HELIOSPRING/target.h [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+#define TARGET_IO_PORTB         0xffff
+#define TARGET_IO_PORTC         0xffff
+#define TARGET_IO_PORTD         (BIT(2))
+
+#define USABLE_TIMER_CHANNEL_COUNT 7
+#define USED_TIMERS             ( TIM_N(1) | TIM_N(8) | TIM_N(4) | TIM_N(12) )
+
+#define IMUF_BIT_I2C_IF_DIS              (1 << 4)
+#ifndef DEBUG
+#define USE_QUAD_MIXER_ONLY
```
Comment (rs2k) [inline src/main/target/HELIOSPRING/target.h]: not sure why this doesn't work in debug mode. :(

File: src/test/unit/scheduler_unittest.cc [hunk 6]
```diff
@@ -89,7 +89,7 @@ extern "C" {
         [TASK_ATTITUDE] = {
             .taskName = "ATTITUDE",
             .taskFunc = imuUpdateAttitude,
-            .desiredPeriod = TASK_PERIOD_HZ(100),
+            .desiredPeriod = TASK_PERIOD_HZ(1000),
```
Comment (rs2k) [inline src/test/unit/scheduler_unittest.cc]: :)

File: src/main/drivers/accgyro/accgyro_imuf9001.c [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+#include "drivers/exti.h"
+#include "drivers/io.h"
+#include "drivers/sensor.h"
+#include "drivers/time.h"
+#include "fc/config.h"
+
+#include "sensors/boardalignment.h"
+volatile int isImufCalibrating = 0;
+
+void crcConfig(void)
```
Comment (rs2k) [inline src/main/drivers/accgyro/accgyro_imuf9001.c]: only used on F3's and F7's. Left it as a place holder for them.

File: src/main/drivers/accgyro/accgyro_imuf9001.c [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+uint8_t imuf9001SpiDetect(const gyroDev_t *gyro)
+{
+    static bool hardwareInitialised = false;
+
+    if (hardwareInitialised) {
+        return(0);
+    }
+
+    crcConfig();
+    //making pancakes
```
Comment (rs2k) [inline src/main/drivers/accgyro/accgyro_imuf9001.c]: what about sausages?

File: src/main/drivers/accgyro/accgyro_mpu.c (hunk 1 above)
Comment (apocolipse) [reply src/main/drivers/accgyro/accgyro_mpu.c]: Will remove

File: src/main/interface/settings.c [hunk 9]
```diff
@@ -96,15 +96,14 @@
 // sync with accelerationSensor_e
 const char * const lookupTableAccHardware[] = {
     "AUTO", "NONE", "ADXL345", "MPU6050", "MMA8452", "BMA280", "LSM303DLHC",
-    "MPU6000", "MPU6500", "MPU9250", "ICM20601", "ICM20602", "ICM20608G", "ICM20649", "ICM20689",
-    "BMI160", "FAKE"
+    "MPU6000", "MPU6500", "MPU9250", "ICM20601", "ICM20602", "ICM20608", "ICM20649", "ICM20689",
+    "BMI160", "ACC_IMUF9001", "FAKE"
 };
```
Comment (apocolipse) [inline src/main/interface/settings.c]: should be ICM20608G

**Pull request merged**